exporter/azuremonitorexporter/log_to_envelope.go

```diff
@@ -55,11 +53,8 @@ func (packer *logPacker) LogRecordToEnvelope(logRecord plog.LogRecord) *contract
 
 	messageData.Message = logRecord.Body().StringVal()
 
-	hexTraceID := logRecord.TraceID().HexString()
-	messageData.Properties[traceIDTag] = hexTraceID
-	envelope.Tags[contracts.OperationId] = hexTraceID
-
-	messageData.Properties[spanIDTag] = logRecord.SpanID().HexString()
+	envelope.Tags[contracts.OperationId] = logRecord.TraceID().HexString()
+	envelope.Tags[contracts.OperationParentId] = logRecord.SpanID().HexString()
```

Comment:
"SpanId" is not "OperationParentId". It's a current trace span.

Reply:
SpanId is the span id of the active Span, when the log was made. It should be stored as operation parentid for correct correlation. The mapping is documented here : https://docs.microsoft.com/en-us/azure/azure-monitor/app/correlation#correlation-headers-using-w3c-tracecontext

(See the existing mapping in ApplicationInsights official sdks https://github.com/microsoft/ApplicationInsights-dotnet/blob/main/BASE/src/Microsoft.ApplicationInsights/Extensibility/OperationCorrelationTelemetryInitializer.cs#L68)

Reply:
See the mapping in the official AzureMonitor exporter:
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryItem.cs#L48-L56

Reply:
See the bullet #1 in the previous comment.
SpanId is the span id of the active Span, when the log was made - exactly. It's not span's parent span.

According to https://docs.microsoft.com/en-us/azure/azure-monitor/app/correlation#correlation-headers-using-w3c-tracecontext:
  - [w3c `parent-id`](https://w3c.github.io/trace-context/#parent-id)=Otel `span-id`=Application Insights `Id`
  - [w3c `trace-id`](https://w3c.github.io/trace-context/#trace-id)=Application Insights `Operation_Id`
  - [w3c `parent-id`](https://w3c.github.io/trace-context/#parent-id) **of this span's parent span**=Application Insights `Operation_ParentId`